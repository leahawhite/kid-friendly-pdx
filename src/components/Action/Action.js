import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default function Action(props) {
    const { name, link, icon, text} = props
    
    return (
      <a target="_blank" rel="noopener noreferrer" href={link}>
        <div className={`place-${name}`}>
          <FontAwesomeIcon icon={icon} size="lg" />
          <p>{text}</p>
        </div>
      </a>
    )
}