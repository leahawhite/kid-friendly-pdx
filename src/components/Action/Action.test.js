import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Action from './Action'

describe('Action component', () => {
  it('renders the UI given props', () => {
    const wrapper = shallow(<Action name="phone" link="tel:+1-555-2222" icon="phone" text="Call" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
