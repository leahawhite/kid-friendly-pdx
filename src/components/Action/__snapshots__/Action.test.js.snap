// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action component renders empty without props 1`] = `
<a
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="place-undefined"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={null}
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <p />
  </div>
</a>
`;

exports[`Action component renders the UI given props 1`] = `
<a
  href="tel:+1-555-2222"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="place-phone"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="phone"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <p>
      Call
    </p>
  </div>
</a>
`;
